File: test_commits/python/feature_addition_original5.py
class Task:
    def __init__(self, name):
        self.name = name
        self.completed = False

    def mark_complete(self):
        self.completed = True

class Project:
    def __init__(self, title):
        self.title = title
        self.tasks = []

    def add_task(self, name):
        self.tasks.append(Task(name))

    def get_completed_tasks(self):
        return [task for task in self.tasks if task.completed]

class ProjectManager:
    def __init__(self):
        self.projects = []

    def add_project(self, title):
        self.projects.append(Project(title))

    def get_all_projects(self):
        return self.projects

File: test_data_do_not_alter/javascript/feature_addition_original5.js
class Task {
    constructor(name) {
        this.name = name;
        this.completed = false;
    }

    markComplete() {
        this.completed = true;
    }
}

class Project {
    constructor(title) {
        this.title = title;
        this.tasks = [];
    }

    addTask(name) {
        this.tasks.push(new Task(name));
    }

    getCompletedTasks() {
        return this.tasks.filter(task => task.completed);
    }
}

class ProjectManager {
    constructor() {
        this.projects = [];
    }

    addProject(title) {
        this.projects.push(new Project(title));
    }

    getAllProjects() {
        return this.projects;
    }
}
